tests/test_mailer.py

```diff
@@ -637,6 +637,32 @@ def test_message_log(self):
             self.assertEqual(log.to_addresses, [])
             self.assertEqual(log.subject, "")
 
+    def test_message_log_without_log_message_data(self):
+        with self.settings(
+            MAILER_EMAIL_BACKEND="django.core.mail.backends.locmem.EmailBackend",
+            MAILER_EMAIL_LOG_MESSAGE_DATA=False,
+        ):  # noqa
+            mailer.send_mail("Subject Log", "Body", "log1@example.com", ["1gol@example.com"])
+
+            self.assertEqual(Message.objects.count(), 1)
+            self.assertEqual(Message.objects.deferred().count(), 0)
+            self.assertEqual(MessageLog.objects.count(), 0)
+            when_added = Message.objects.first().when_added
+
+            engine.send_all()
+
+            self.assertEqual(Message.objects.count(), 0)
+            self.assertEqual(Message.objects.deferred().count(), 0)
+            self.assertEqual(MessageLog.objects.count(), 1)
+
+            log = MessageLog.objects.all()[0]
+
+            self.assertEqual(log.email, None)
+            self.assertEqual(log.message_data, None)
+            self.assertEqual(log.to_addresses, [])
+            self.assertEqual(log.subject, "")
```

Comment:
I feel like it would be more useful and correct for these to both return `None` for this case. If we change them to `None`, it means that any other code that attempts to use these fields as if data was present is more likely to raise errors, which is a good thing. 

In fact, what about making them raise exceptions? Any code that is attempting to get `subject` or `to_addresses` of a `MessageLog`, and also needs to deal with the possibility of `email` being `None`, needs to be updated with `if email is not None` or similar. So I suspect the best would be to make these properties raise exceptions. But I don't know what other consequences that might have.

Reply:
Having looked into it more, I realised the existing `db_to_email` to email method returns `None` for a bunch of error cases, and the existing code does a load of `if email is not None`.

I'm not convinced this is actually the most useful behaviour, but we should maintain that approach. So I'm fine with this. Sorry for the confusion!

Reply:
Thanks for having a look and merging.

Do you think it is worth a new release, maybe together with https://github.com/pinax/django-mailer/pull/173?

Reply:
Released now.